modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java

```diff
@@ -3753,134 +3795,98 @@ else if (log.isDebugEnabled())
             else
                 U.writeFully(ch, ByteBuffer.wrap(U.IGNITE_HEADER));
 
-            ClusterNode locNode = getLocalNode();
-
-            if (locNode == null)
-                throw new IgniteCheckedException("Local node has not been started or " +
-                    "fully initialized [isStopping=" + getSpiContext().isStopping() + ']');
-
-            if (recovery != null) {
-                HandshakeMessage msg;
-
-                int msgSize = HandshakeMessage.MESSAGE_FULL_SIZE;
-
-                if (handshakeConnIdx != null) {
-                    msg = new HandshakeMessage2(locNode.id(),
-                        recovery.incrementConnectCount(),
-                        recovery.received(),
-                        handshakeConnIdx);
-
-                    msgSize += 4;
-                }
-                else {
-                    msg = new HandshakeMessage(locNode.id(),
-                        recovery.incrementConnectCount(),
-                        recovery.received());
-                }
-
-                if (log.isDebugEnabled())
-                    log.debug("Writing handshake message [locNodeId=" + locNode.id() +
-                        ", rmtNode=" + rmtNodeId + ", msg=" + msg + ']');
+            // Step 2. Prepare Handshake message to send remote node.
```

Comment:
Is it easy to separate each step into a private method? In that case code of the `safeTcpHandshake` would become self-documented.

Reply:
Sergey, I've tried to avoid huge changes and focused only on removing not used code.
In general, I agree with you here - better to have separate methods on each step of handshaking process, but in fact, this method used only in the signle flow and single time and we can leave it as it is.
WDYT?